### PR TITLE
Use ParameterizedMethodType instead of Pair

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

**Comment by mernst** [comment]
@wmdietl The Travis job will pass when merged as you can see from [the mfu-type branch here](https://travis-ci.org/mernst/checker-framework/branches).  It's failing in this pull request because it's looking for a branch of checker-framework-inference named `mfu-type`, and that is in the `mernst` organization rather than the `typetools` organization.  (This indicates that in the future, these coordinated pull requests should be from branches in the `typetools` organization when possible.)

TL;DR this is ready for review despite the (spurious) Travis failure.

File: checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package org.checkerframework.checker.guieffect;
 
+import static org.checkerframework.framework.type.AnnotatedTypeFactory.ParameterizedMethodType;
```
Comment (smillst) [inline checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java]: It's a bit strange that this is statically imported

**Comment by mernst** [comment]
@wmdietl I merged this because it got one approval, you are traveling, and other pull requests require it.  We can revisit the class and field names later if you have better suggestions.

File: checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java (hunk 1 above)
Comment (wmdietl) [reply checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java]: Overall, replacing the `Pair` with a dedicated class is definitely good.
I also don't like the static imports. Are they needed because `ParameterizedMethodType` is an inner class?

Comment (mernst) [reply checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java]: > Are they needed because ParameterizedMethodType is an inner class?

Yes.  The alternative is writing `AnnotatedTypeFactory.ParameterizedMethodType` everywhere that `ParameterizedMethodType` is currently used.

Comment (smillst) [reply checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java]: I thought static imports where for importing methods and fields -- not classes.  The code compiles even when static is removed.

Comment (wmdietl) [reply checker/src/main/java/org/checkerframework/checker/guieffect/GuiEffectVisitor.java]: I agree with @smillst . The `static` isn't needed and the code would be nicer without.
